editor/src/components/canvas/ui-jsx-canvas.tsx

```diff
@@ -571,20 +576,54 @@ export const UiJsxCanvas = React.memo<UiJsxCanvasPropsWithErrorCallback>((props)
       style={{
         all: 'initial',
       }}
+      id={CanvasContainerShadowRoot}
+      data-testid={CanvasContainerShadowRoot}
     >
-      <Helmet>{parse(linkTags)}</Helmet>
-      <RerenderUtopiaCtxAtom.Provider value={rerenderUtopiaContextValue}>
-        <UtopiaProjectCtxAtom.Provider value={utopiaProjectContextValue}>
-          <CanvasContainer
-            validRootPaths={rootValidPathsArray}
-            canvasRootElementElementPath={storyboardRootElementPath}
-          >
-            <SceneLevelUtopiaCtxAtom.Provider value={sceneLevelUtopiaContextValue}>
-              {StoryboardRoot}
-            </SceneLevelUtopiaCtxAtom.Provider>
-          </CanvasContainer>
-        </UtopiaProjectCtxAtom.Provider>
-      </RerenderUtopiaCtxAtom.Provider>
+      <ReactShadowRoot declarative={props.renderedInSSR}>
+        <EmotionWrapper>
+          <style>{`div,
+          span,
```

Comment:
Why are these elements specifically called out here?

Reply:
Ok good point actually, I need to add a `FIXME` here.
This styling is required, and happened to be applied to the canvas via our editor's css here <https://github.com/concrete-utopia/utopia/blob/master/editor/resources/editor/css/initial-load.css#L58-L67>. At one point it was discovered that certain tests would fail because `border-box` was missing, so it was added to all tests here: <https://github.com/concrete-utopia/utopia/blob/c44e0ac5de859169d0caf25957031ba153a6503e/editor/src/components/canvas/ui-jsx.test-utils.tsx#L606-L614>
But that no longer works because of the shadow root, so we now need to include that border-box styling here inside the shadow-root. Really we shouldn't be including this at all, but that's a separate issue to fix (hence why I need to add a `FIXME` here)